src/pybind/mgr/orchestrator.py

```diff
@@ -209,103 +567,14 @@ def copy_to_this_subinterpreter(r_obj):
         raise e
 
 
-class ReadCompletion(_Completion):
-    """
-    ``Orchestrator`` implementations should inherit from this
-    class to implement their own handles to operations in progress, and
-    return an instance of their subclass from calls into methods.
-    """
-
-    def __init__(self):
-        pass
-
-    @property
-    def is_read(self):
-        return True
-
-    @property
-    def should_wait(self):
-        """Could the external operation be deemed as complete,
-        or should we wait?
-        We must wait for a read operation only if it is not complete.
-        """
-        return not self.is_complete
-
-
-class TrivialReadCompletion(ReadCompletion):
+class TrivialReadCompletion(Completion):
     """
     This is the trivial completion simply wrapping a result.
     """
     def __init__(self, result):
         super(TrivialReadCompletion, self).__init__()
         self._result = result
```

Comment:
There's a problem with TrivalReadCompletions - `has_result` never returns true, so if one of these objects turns up, the orchestrator CLI will spin logging "mgr[orchestrator_cli] Operations pending: 1" forever. I thought maybe setting `self._value = result` instead might do it, but it's not enough. This seems to work though:

```suggestion
        self.finalize(result)
```

Reply:
Just to add to the above, this was found through manual testing the DeepSea orchestrator. The first time running `ceph orchestrator device ls` or `ceph orchestrator service ls` ran fine, the second time it would hang, due to the never-finalized TrivialReadCompletion. AFAICT nothing else is using TrivialReadCompletion, expect selftest/module.py, but in that context the completions are explicitly `fail()`ed, so the problem never manifested. I've opened https://github.com/sebastian-philipp/ceph/pull/4 to fix it.